leave the yaml at home post

krancour opened the pull request:
Pinging the usual suspects for review (@vdice, @karenhchu, @bridgetkromhout, and @squillace) as well as @jorgearteiro and @devigned -- both of whom, whether intentionally or otherwise, inspired me to write this.

arschles commented on content/2022-05-19-leave-the-yaml-at-home.md:
````diff
@@ -0,0 +1,185 @@
+console.log("Hello, World!")
+```
+
+Or could mux events, dispatching them to the correct handler function:
````
```suggestion
Or could multiplex events, dispatching them to the correct handler function:
```

krancour replied:
I think I might end up not using either of these two words...

When I think of muxing/multiplexing, I'm usually thinking of a long-running process like a web server that dispatches concurrent work in-process on the basis of path, header, etc.

Here, the worker isn't long-running. It fires up and handles a _single_ event.

So on a technicality, this probably isn't multiplexing at all, even though it looks and smells like it.

I think I might just go with:

> Or could dispatch an event to the correct handler function:

arschles replied:
yea, `dispatch` sounds good there 👍

arschles commented on content/2022-05-19-leave-the-yaml-at-home.md:
```diff
@@ -0,0 +1,185 @@
+`make test-unit` and `make lint` concurrently (subject to scheduling
+constraints). Each executes in its own container, in its own pod. Those
+containers are both based on a `brigadecore/go-tools:v0.9.0` image that's
+pre-loaded with all of our team's most frequently used Go-based tools.
```
just in case people don't know what `Go` refers to:

```suggestion
pre-loaded with all of our team's most frequently used [Go](https://golang.org)-based tools.
```

arschles commented on content/2022-05-19-leave-the-yaml-at-home.md:
```diff
@@ -0,0 +1,185 @@
+
+With JavaScript and TypeScript being
+[Turing complete](https://en.wikipedia.org/wiki/Turing_completeness) languages,
+the full range of what you can accomplish with Brigade is unbounded. More than
+simply being unbounded, you're free to implement your workflows using everything
```
```suggestion
the full range of what you can accomplish with Brigade is unbounded. Within a Brigade script, for example, you're free to implement your workflows using everything
```

arschles commented on content/2022-05-19-leave-the-yaml-at-home.md:
```diff
@@ -0,0 +1,185 @@
+manager. Whenever applicable, Brigade workers will run an `npm install` or
+`yarn install` before executing your script!
+
+With all this power and flexibility at your finger tips, isn't it time you left
+the YAML at home?
```
I think this sounded slightly less like a sales pitch:

```suggestion
With all the power of a full-featured programming language, we think it's time to leave YAML at home and start writing Brigade scripts.
```

arschles commented on content/2022-05-19-leave-the-yaml-at-home.md:
```diff
@@ -0,0 +1,185 @@
+
+We've heard from time to time that people aren't always so crazy about YAML --
+and that's putting it nicely. I don't personally harbor such strong anti-YAML
+sentiment, but the critics are out there and as critics are wont to be, they're
+quite vocal. If you dare to search for "yaml sucks" in your favorite search
```
```suggestion
sentiment, but critics are out there, and they're quite vocal. If you dare to search for "yaml sucks" in your favorite search
```

krancour replied:
Yup.... "wont" is too archaic. Changing.

arschles commented on content/2022-05-19-leave-the-yaml-at-home.md:
```diff
@@ -0,0 +1,185 @@
+and that's putting it nicely. I don't personally harbor such strong anti-YAML
+sentiment, but the critics are out there and as critics are wont to be, they're
+quite vocal. If you dare to search for "yaml sucks" in your favorite search
+engine, you'll find a litany of criticism -- some of it valid. So we thought
```
I think you should briefly describe why some of the criticism is valid. Otherwise, I don't think the `some of it valid` part adds much.

krancour replied:
I think I was just trying to neither invalidate nor buy into the (general) criticism of YAML. I don't hate YAML, but I'm sure those who do make some solid points.

I think this comment has actually given me a lot to think about... If I don't actually hate YAML so much, why do I see our minimal YAML as a selling point? I guess my _real_ problem is that I don't see a declarative language as expressive as an imperative one for modeling workflows.

I think I need to sell that idea a little harder without latching on to more generalized anti-YAML sentiment.

krancour commented on content/2022-05-19-leave-the-yaml-at-home.md:
```diff
@@ -0,0 +1,185 @@
+
+We've heard from time to time that people aren't always so crazy about YAML --
+and that's putting it nicely. I don't personally harbor such strong anti-YAML
+sentiment, but the critics are out there and as critics are wont to be, they're
```
https://www.merriam-webster.com/dictionary/wont

Do people not use this? I feel like this is where @bridgetkromhout would be reminding me to use the simplest language I can get away with.

I'm going to rephrase this is as:

> ... but the critics are out there, and as critics often are, they're quite vocal.

carolynvs replied:
I use that term but yeah it's not well known for a lot of people

krancour replied:
@carolynvs I feel better knowing I'm not the only one!!!

arschles commented on content/2022-05-19-leave-the-yaml-at-home.md:
```diff
@@ -0,0 +1,185 @@
+      field to indicate where project source -- including the script we'd like
+      to run -- can be obtained.
+
+This is all the YAML you write! (And, actually, most of this can be generated
+for you with the `brig init` command!)
```
I think the parentheses should go. That seems like a really important point that maybe even deserves a little bit more prominence.

krancour replied:
sgtm

arschles reviewed (commented):
lots of comments here. most are nits but I did my best to add clarity and conciseness where possible.

carolynvs commented:
I was initially confused because this says leave the yaml at home, and that brigade doesn't use much yaml then immediately starts breaking down and documenting a yaml file.

I think it would be more clear and not derail the point you are making here by omitting the k8s manifest and instead say along the lines of "Brigade runs on kubernetes, so a kubernetes manifest is required to install Brigade, but everything in Brigade uses javascript or typescript, so after installing you are YAML free", and then just move on to the rest of your blog post.

This would really tighten up the post and get you to the point a lot quicker and hopefully reduce how many bounce at the manifest part.

krancour commented:
> I think it would be more clear and not derail the point you are making here by omitting the k8s manifest and instead say along the lines of "Brigade runs on kubernetes, so a kubernetes manifest is required to install Brigade, but everything in Brigade uses javascript or typescript, so after installing you are YAML free", and then just move on to the rest of your blog post.

That YAML you saw wasn't part of the installation process. It's how projects are defined. Projects are defined declaratively. Logic is defined imperatively.

But I see your point that for a post that talks about leaving the YAML at home, it sure does spend a lot of time dissecting a chunk of YAML -- and I definitely need to fix that somehow.

I am going to experiment with dissecting it less and being a bit hand-wavy about the specifics of the project definition -- after all, this post is _not_ mean to be a comprehensive tutorial. The real point I need to get across, and more succinctly is: "Ya... You define your projects using a very small chunk of YAML and that's it. You're done with the YAML now."

carolynvs commented:
ah okay I kind of lump projects into "brigade installation" in my head but I'm not a true brigadier

krancour commented:
@carolynvs I did two things in response to your comment:

1. I added a "pullout quote" _before_ the YAML to add more impact to the point that you only write a small amount of YAML.

2. I removed the detailed dissection of the YAML and the post doesn't seem to suffer as a result.

Thanks for the great advice!
